### PR TITLE
Enable cadl ranch for modular

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1919

### CODE REVIEW

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 1]
```diff
@@ -21,11 +21,17 @@ function getRLCResponseType(operation: Operation) {
     new Set(operation.responses.flatMap((r) => r.statusCodes)).values()
   );
 
-  statusCodes.push("Default");
+  if (hasRLCDefaultResponse(operation)) {
+    statusCodes.push("Default");
+  }
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:26]: I don't think we need this if you merge main, we have already mapped the all rlc responses type to modular operations.

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 2]
```diff
@@ -99,6 +107,9 @@ export function getDeserializePrivateFunction(
       `}`
     );
   } else if (returnType.type === "void") {
+    if (statements.length === 0) {
+      statements.push(`result;`);
+    }
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:112]: what is this trying to resolve ?

File: packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/UsageClient.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+export class UsageClient {
+  private _client: UsageContext;
+
+  /** Illustrates usage of Record in different places(Operation parameters, return type or both). */
+  constructor(options: UsageClientOptions = {}) {
+    this._client = createUsage(options);
+  }
+
+  /** */
```
Comment (qiaozha) [inline packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/UsageClient.ts:29]: Is this normal ?

File: packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/models/models.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/** Record used in operation return type */
+export interface OutputRecord {
+  /** */
```
Comment (qiaozha) [inline packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/models/models.ts:6]: The same here

File: packages/typespec-ts/test/modularIntegration/usage.spec.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import { assert } from "chai";
+import {
+  UsageContext,
+  createUsage,
+  input,
+  inputAndOutput,
+  output
+} from "./generated/models/usage/generated/src/api/index.js";
+const EXPECTED_VALUE = "example-value";
+describe("UsageContext Classical Client", () => {
```
Comment (qiaozha) [inline packages/typespec-ts/test/modularIntegration/usage.spec.ts:11]: kind of feel like it's not necessary to test every api works both at classical client layer and the api layer in the future. Cadl ranch won't calculate them separately.

Comment (MaryGao) [reply packages/typespec-ts/test/modularIntegration/usage.spec.ts:11]: Yeah I think so, we could only test for classical layer only in future considering the classical would call the api layer.

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts (hunk 2 above)
Comment (MaryGao) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts:112]: https://github.com/Azure/autorest.typescript/pull/1931/files#diff-3d6099d37e97ef4ac1cb7f3b96c64afb5f22edfccbeda5560dc70e0474bf7e62R34-R39

File: packages/typespec-ts/test/modularIntegration/usage.spec.ts (hunk 5 above)
Comment (qiaozha) [reply packages/typespec-ts/test/modularIntegration/usage.spec.ts:11]: I am okay with it.

File: packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/UsageClient.ts (hunk 3 above)
Comment (MaryGao) [reply packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/UsageClient.ts]: resolved

File: packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/models/models.ts (hunk 4 above)
Comment (MaryGao) [reply packages/typespec-ts/test/modularIntegration/generated/models/usage/generated/src/models/models.ts]: fixed

File: packages/typespec-ts/test/commands/cadl-ranch-list.ts [hunk 6]
```diff
@@ -150,3 +150,11 @@ export const cadls: CadlRanchConfig[] = [
     inputPath: "azure/core/traits"
   }
 ];
+
+export const modularTsps: TypeSpecRanchConfig[] = [
+  {
+    outputPath: "models/usage",
+    inputPath: "type/model/usage",
+    debug: true
```
Comment (qiaozha) [inline packages/typespec-ts/test/commands/cadl-ranch-list.ts:158]: should remove this ?

**Review by qiaozha — APPROVED** [review]
LGTM, just a minor comment.

**Pull request merged**